hazelcast-client-new/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -123,7 +123,31 @@ protected AbstractClientInternalCacheProxy(CacheConfig cacheConfig, ClientContex
         syncListenerRegistrations = new ConcurrentHashMap<CacheEntryListenerConfiguration, String>();
         syncLocks = new ConcurrentHashMap<Integer, CountDownLatch>();
 
-        NearCacheConfig nearCacheConfig = cacheConfig.getNearCacheConfig();
+        NearCacheConfig clientNearCacheConfig = clientContext.getClientConfig().getNearCacheConfig(name);
```

Comment:
This method is getting too long. Can you check if checkstyle is skipping this file?